addons/safemode/functions/fnc_lockSafety.sqf

```diff
@@ -60,7 +60,13 @@ if (_unit getVariable [QGVAR(actionID), -1] == -1) then {
 };
 
 if (_muzzle isEqualType "") then {
+    private _laserEnabled = _unit isIRLaserOn _weapon || {_unit isFlashlightOn _weapon};
```

Comment:
No need for an extra variable.

Reply:
There is need. Laser is disabled right after `selectWeapon` and we need to check if it was enabled.

Reply:
You only use this condition once, you do not need to store it.

Reply:
I store state not condition. Or I don't understand you.

Reply:
this is a local var that is only used once (the check below) and then is destroyed.

Reply:
>You only use this condition once, you do not need to store it.

You cannot evaluate the condition after LIne 65 though. So how else would you do it?

Reply:
true, I may or may not have seen that selectWeapon ...
wooops.